torch/distributed/distributed_c10d.py

```diff
@@ -965,6 +965,8 @@ def _get_default_store():
 
 def _update_default_pg(pg):
     _world.default_pg = pg
+    rank = pg.rank() if pg is not None else -1
```

Comment:
We should also check if pg is `GroupMember.NON_GROUP_MEMBER` ad set to -1 if the current rank is not in the group.